package/src/components/Message/MessageSimple/MessageEditedTimestamp.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+import { MessageTimestamp, MessageTimestampProps } from './MessageTimestamp';
+
+import {
+  MessageContextValue,
+  useMessageContext,
+} from '../../../contexts/messageContext/MessageContext';
+import { useTheme } from '../../../contexts/themeContext/ThemeContext';
+import { useTranslationContext } from '../../../contexts/translationContext/TranslationContext';
+import { DefaultStreamChatGenerics } from '../../../types/types';
+import { isEditedMessage } from '../../../utils/utils';
+
+export type MessageEditedTimestampPropsWithContext<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = Pick<MessageContextValue<StreamChatGenerics>, 'message'> & MessageTimestampProps;
+
+export const MessageEditedTimestampWithContext = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  props: MessageEditedTimestampPropsWithContext<StreamChatGenerics>,
+) => {
+  const { message, timestamp } = props;
+  const {
+    theme: {
+      colors: { grey },
+      messageSimple: {
+        content: { editedLabel, editedTimestampContainer },
+      },
+    },
+  } = useTheme();
+  const { t } = useTranslationContext();
+
+  if (!isEditedMessage(message)) {
+    return null;
+  }
+
+  return (
+    <View style={[styles.container, editedTimestampContainer]}>
+      <Text style={[styles.text, { color: grey }, editedLabel]}>{t<string>('Edited ')}</Text>
+      <MessageTimestamp calendar={true} timestamp={timestamp} />
+    </View>
+  );
+};
+
+const areEqual = <StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics>(
+  prevProps: MessageEditedTimestampPropsWithContext<StreamChatGenerics>,
+  nextProps: MessageEditedTimestampPropsWithContext<StreamChatGenerics>,
+) => {
+  const { message: prevMessage } = prevProps;
+  const { message: nextMessage } = nextProps;
+  const messageEqual = prevMessage.message_text_updated_at === nextMessage.message_text_updated_at;
+  if (!messageEqual) return false;
+
+  return true;
+};
+
+const MemoizedMessageEditedTimestamp = React.memo(
+  MessageEditedTimestampWithContext,
+  areEqual,
+) as typeof MessageEditedTimestampWithContext;
```

Comment:
Are there any good reasons why we are memorizing this component? All I can see are troubles with stale data down the road.

Reply:
There are no reasons. This has been a pattern in our RN SDK for a long time, and it's just been followed here as well. This way, we don't pass props down the hierarchy but can utilize them from context(if present). These are things that have to be improved everywhere explicitly.

Reply:
I think excessive memoization creates more problems than benefits. I understand this is a legacy pattern we also have in our React SDK but we should slowly move away from it (at least for the new stuff). Otherwise, we keep increasing the technical dept and we are breaking the expectations of React components - re-render when a prop changes.

Reply:
Also I see a problem with this memoisation, what if the timestamp changes? 

we can remove this memoisation all together, there seems to be nothing expensive in the render function that warrants for memoisation here

Reply:
I have removed it completely